Source/JavaScriptCore/runtime/TemporalDuration.cpp

```diff
@@ -304,6 +319,43 @@ static TemporalUnit largestSubduration(const ISO8601::Duration& duration)
     return static_cast<TemporalUnit>(index);
 }
 
+// https://tc39.es/proposal-temporal/#sec-temporal-timedurationfromcomponents
+Int128 TemporalDuration::timeDurationFromComponents(double hours, double minutes, double seconds, double milliseconds, double microseconds, double nanoseconds)
+{
+    Int128 min = static_cast<Int128>(minutes) + static_cast<Int128>(hours) * 60;
+    Int128 sec = static_cast<Int128>(seconds) + min * 60;
+    Int128 millis = static_cast<Int128>(milliseconds) + sec * 1000;
+    Int128 micros = static_cast<Int128>(microseconds) + millis * 1000;
+    Int128 nanos = static_cast<Int128>(nanoseconds) + micros * 1000;
```

Comment:
I think we should use `checkedCastDoubleToInt128` to safely add, similar to `Duration::totalNanoseconds`

Reply:
Thanks for the suggestion. I'll check if this current revision builds.